src/keyring/KeyringController.test.ts

```diff
@@ -230,6 +288,26 @@ describe('KeyringController', () => {
     ).rejects.toThrow('Key derivation failed - possibly wrong passphrase');
   });
 
+  it('should import account with strategy json empty password', async () => {
+    await expect(
+      keyringController.importAccountWithStrategy(AccountImportStrategy.json, [
+        input,
+        '',
+      ]),
+    ).rejects.toThrow('Key derivation failed - possibly wrong passphrase');
+  });
+
+  /**
+   * If there is only HD Key Tree keyring with 1 account and removeAccount is called passing that account
+   * It deletes keyring object also from state - not sure if this is correct behavior.
```

Comment:
Interesting. Right now we don't make it possible to delete these accounts, so we never encountered this. I agree that it seems like the wrong thing to do.